ssl/quic/quic_impl.c

```diff
@@ -605,6 +616,57 @@ static void qc_set_default_xso(QUIC_CONNECTION *qc, QUIC_XSO *xso, int touch)
         SSL_free(&old_xso->ssl);
 }
 
+/* SSL_set_options */
+QUIC_TAKES_LOCK
+static uint64_t quic_mask_or_options(SSL *ssl, uint64_t mask_value, uint64_t or_value)
+{
+    QCTX ctx;
+    uint64_t options;
+
+    if (!expect_quic(ssl, &ctx))
+        return 0;
+
+    quic_lock(ctx.qc);
+
+    /*
+     * Currently most options that we permit are handled in the handshake
+     * layer.
+     */
+    or_value &= OSSL_QUIC_PERMITTED_OPTIONS;
+
+    SSL_clear_options(ctx.qc->tls, mask_value);
+    options = SSL_set_options(ctx.qc->tls, or_value);
```

Comment:
Should these be called only if this is a true connection object? When application wants to set/clear SSL_OP_CLEANSE_PLAINTEXT, it does not IMO want to affect the TLS options. Although currently this is fairly harmless because when streams are in use the handshake is already completed so any changes of TLS options probably won't have any effect anyway.

Reply:
Yeah, this needs a rethink.